framework/common/encryption.cpp

```diff
@@ -134,6 +135,14 @@ bool diffie_hellman::compute_key(uint8_t *key, unsigned &key_length, const uint8
     return true;
 }
 
+void diffie_hellman::copy_pubkey(uint8_t *dest) const
```

Comment:
That's not really diffie hellman related logic.
Perhaps it can be moved into a generic "copy with padding" (left or right) function in one of the utility classes.

Reply:
Agree, but I think its rare that we'll need such a function, so I guess this should be a (nitpick)

Reply:
The idea was that this would be the only way to access the pubkey, i.e. the pubkey() method would be removed. Annoyingly, though, it is (or rather was) used for debug output.

Reply:
So, If I'd make an external function `copy_pubkey(const diffie_helman &dh, uint8_t *dest)`, you would be happy?

Reply:
Agree on rare/limited use, so IMO should not be  a blocker for merging.

Reply:
Moved to global instead of diffie_hellman member.